### PR TITLE
fix(l18n): react to only be peer dependency

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

I am debugging a very odd bug with redux and react-redux and invalid hook calls. This PR suggests to mainly remove react as a direct dependency from a package. As the error can somewhat only come from duplicated react versions being bundled from my view.

### CODE REVIEW

File: application-templates/starter/package.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "@commercetools-uikit/spacings": "10.15.1",
     "@commercetools-uikit/text": "10.15.1",
     "prop-types": "15.7.2",
-    "react": "16.13.0",
+    "react": "16.13.1",
```
Comment (tdeekens) [inline application-templates/starter/package.json:27]: Just while here...

Comment (tdeekens) [inline application-templates/starter/package.json:27]: Just while here...

File: packages/l10n/package.json [hunk 2]
```diff
@@ -44,8 +44,7 @@
     "@commercetools-frontend/sentry": "16.5.3",
     "@types/prop-types": "^15.7.3",
     "moment": "^2.24.0",
-    "prop-types": "15.7.2",
-    "react": "16.13.0"
```
Comment (tdeekens) [inline packages/l10n/package.json:48]: Moving this down into devDeps.

Comment (emmenko) [reply packages/l10n/package.json:48]: Ah, it should probably also be a peer dep. Good catch!

Comment (tdeekens) [reply packages/l10n/package.json:48]: React is a devDep mostly. So I presume we add a peerDep requirement here too.

Comment (tdeekens) [reply packages/l10n/package.json:48]: 2e2d854

**Pull request merged**